importer/wdi_importer.py

```diff
@@ -1,63 +1,922 @@
+            category_vars[value['category']].append(key)
+
+    existing_categories = DatasetCategory.objects.values('name')
+    existing_categories_list = []
+
+    for each in existing_categories:
+        existing_categories_list.append(each['name'])
+
+    if wdi_category_name_in_db not in existing_categories_list:
+        the_category = DatasetCategory(name=wdi_category_name_in_db)
+        the_category.save()
+        backup_sql_file.write("INSERT INTO %s (name, created_at, updated_at) VALUES ('%s', '%s', '%s');\n" %
+                              (DatasetCategory._meta.db_table, wdi_category_name_in_db.replace("'", "\\'"),
+                               the_category.created_at.strftime('%Y-%m-%d %H:%M:%S'),
+                               the_category.updated_at.strftime('%Y-%m-%d %H:%M:%S')))
+        logger.info("Inserting a category %s." % wdi_category_name_in_db.encode('utf8'))
+
+    else:
+        the_category = DatasetCategory.objects.get(name=wdi_category_name_in_db)
+
+    existing_subcategories = DatasetSubcategory.objects.filter(fk_dst_cat_id=the_category.pk).values('name')
+    existing_subcategories_list = []
+
+    for each in existing_subcategories:
+        existing_subcategories_list.append(each['name'])
+
+    wdi_categories_list = []
+
+    for key, value in category_vars.items():
+        wdi_categories_list.append(key)
+        if key not in existing_subcategories_list:
+            the_subcategory = DatasetSubcategory(name=key, fk_dst_cat_id=the_category)
+            the_subcategory.save()
+            backup_sql_file.write(
+                "INSERT INTO %s (name, fk_dst_cat_id, created_at, updated_at) VALUES ('%s', %s, '%s', '%s');\n" %
+                (DatasetSubcategory._meta.db_table, key.replace("'", "\\'"), the_category.pk,
+                 the_subcategory.created_at.strftime('%Y-%m-%d %H:%M:%S'),
+                 the_subcategory.updated_at.strftime('%Y-%m-%d %H:%M:%S')))
+            logger.info("Inserting a subcategory %s." % key.encode('utf8'))
+
+    existing_entities = Entity.objects.values('name')
+    existing_entities_list = []
+    for each in existing_entities:
+        existing_entities_list.append(each['name'])
+
+    country_tool_names = CountryName.objects.all()
+    country_tool_names_dict = {}
+    for each in country_tool_names:
+        country_tool_names_dict[each.country_name.lower()] = each.owid_country
+
+    country_name_entity_ref = {}  # this dict will hold the country names from excel and the appropriate entity object (this is used when saving the variables and their values)
+
+    row_number = 0
+    for row in country_ws.rows:
+        row_number += 1
+        for cell in row:
+            if row_number > 1:
+                column_number += 1
+                if column_number == 1:
+                    country_code = cell.value
+                if column_number == 3:
+                    country_name = unidecode.unidecode(cell.value)
```

Comment:
Is the use of unidecode necessary? We should be able to handle unicode strings throughout the grapher-- if we can't then that's a bug that should be fixed.